src/bin/genemichaels.rs

```diff
@@ -244,14 +345,127 @@ fn main() {
                 Ok(_) => { },
                 Err(e) => {
                     if !args.quiet {
-                        eprintln!("Error formatting {}: {:?}", &file.to_string_lossy(), e);
+                        print_error_text();
+                        eprintln!("formatting {}: {:?}", &file.to_string_lossy(), e);
                     }
                     failed = true;
                 },
             };
         }
+        eprintln!(
+            "\x1B[1;32m    Finished\x1B[0;22m workspace formatted successfully in {:.2}s",
+            time::Instant::now().duration_since(inst).as_secs_f64()
+        );
         if failed {
             process::exit(1);
         }
     }
 }
+
+fn process_cargo_toml(path: PathBuf, manifest: Manifest, args: &Args, config: FormatConfig) {
+    let dirs = match manifest.workspace.map(|wkspc| wkspc.members) {
+        Some(mut mb) => {
+            if let Some(bins) = manifest.bin {
```

Comment:
So `manifest.bin` etc refer to the source directories in the "main" project directory rather than within the workspace, so logically they shouldn't be in the `Some` branch of `wkspc.members`.

I think it would be good to make a new Vec here too.  You're restricted by the fact that `members` needs strings rather than paths, and reusing a field for another purpose than it was originally intended would confuse me if I were just jumping in.

Reply:
From my initial testing this wasn't the case and it does work as expected in its current form (at least for `lib`). I'll have to do some more testing but perhaps if you have a project, you've been testing against with that folder structure, it would be prudent to do some testing of your own too.

Reply:
Okay, I took a closer look and I was confused with a couple assumptions I was making about Cargo.toml.  There are some issues though
* The `None` branch is dead code ATM
* If it weren't (i.e. the command accepted workspace-less manifests) it would ignore `[[bin]]` etc paths
* `src` is ignored if all `bin` paths are outside of `src`.  TBH this seems to be an inaccuracy on the cargo.toml parsing library (i.e. `main.rs` in `src/` is always compiled).  Making assumptions again, but I assume this also applies to tests, benches, etc.
* The whole workspace member dir is scanned, even if there are non-project rust files there (like test data files)

Reply:
I think I've got it pretty close to what you were after but please let me know if the current iteration isn't quite working as you expect.